```
Error running mint automation
+ mkdir -p 9203-ba41b59/gopath/src/github.com/minio
+ git -C 9203-ba41b59/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 9203-ba41b59/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 9203-ba41b59/gopath/src/github.com/minio/minio fetch --quiet minio pull/9203/head:pr9203
+ git -C 9203-ba41b59/gopath/src/github.com/minio/minio checkout --quiet pr9203
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/9203-ba41b59/gopath
+ make -C 9203-ba41b59/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/Azure/azure-pipeline-go v0.2.1
go: downloading github.com/minio/minio-go/v6 v6.0.51-0.20200319192131-097caa7760c7
go: downloading github.com/json-iterator/go v1.1.9
go: downloading github.com/klauspost/compress v1.10.3
go: downloading github.com/tinylib/msgp v1.1.1
go: downloading github.com/minio/sio v0.2.0
go: downloading golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/miekg/dns v1.1.8
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/rs/cors v1.6.0
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/shirou/gopsutil v2.18.12+incompatible
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: downloading github.com/gorilla/handlers v1.4.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: downloading golang.org/x/crypto v0.0.0-20191117063200-497ca9f6d64f
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: downloading github.com/klauspost/cpuid v1.2.2
go: downloading github.com/minio/parquet-go v0.0.0-20200125064549-a1e49702e174
go: downloading github.com/prometheus/common v0.2.0
go: downloading github.com/klauspost/pgzip v1.2.1
go: downloading go.uber.org/atomic v1.3.2
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/mattn/go-isatty v0.0.7
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/gorilla/mux v1.7.0
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/philhofer/fwd v1.0.0
go: downloading github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/Shopify/sarama v1.24.1
go: downloading google.golang.org/grpc v1.22.0
go: downloading github.com/beevik/ntp v0.2.0
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/golang/protobuf v1.3.2
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.2.3
go: downloading github.com/gogo/protobuf v1.2.1
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: downloading github.com/pkg/errors v0.8.1
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/minio/lsync v1.0.1
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/hashicorp/vault/api v1.0.4
go: downloading github.com/tidwall/gjson v1.3.5
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/nats-io/stan.go v0.4.5
go: downloading github.com/tidwall/pretty v1.0.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading go.opencensus.io v0.21.0
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading golang.org/x/text v0.3.2
go: downloading gopkg.in/ini.v1 v1.48.0
go: downloading github.com/cheggaaa/pb v1.0.28
go: downloading github.com/klauspost/reedsolomon v1.9.3
go: downloading github.com/mattn/go-colorable v0.1.1
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: downloading google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/nats-io/nats.go v1.9.1
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading gopkg.in/yaml.v2 v2.2.4
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/nats-io/jwt v0.3.2
go: downloading gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/lib/pq v1.1.1
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/nats-io/nkeys v0.1.3
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading github.com/tidwall/sjson v1.0.4
# github.com/minio/minio/cmd
cmd/xl-sets.go:185:6: onlineDisks declared but not used
make: *** [build] Error 2

```

--- a/cmd/xl-sets.go
+++ b/cmd/xl-sets.go
@@ -182,29 +182,51 @@ func findDiskIndex(refFormat, format *formatXLV3) (int, int, error) {
 // connectDisksWithQuorum is same as connectDisks but waits
 // for quorum number of formatted disks to be online in any given sets.
 func (s *xlSets) connectDisksWithQuorum() {
-	var onlineDisks int
-	diskMap := s.getDiskMap()
+	diskMap := make(map[string]StorageAPI)
+	var diskMapMu sync.RWMutex
 
-	for onlineDisks < len(s.endpoints)/2 {
-		for i, endpoint := range s.endpoints {
-			if isEndpointConnected(diskMap, s.endpointStrings[i]) {
+	connectQ := make(chan struct{}, len(s.endpoints)/2)
+
+	for {
+		for epIdx := range s.endpoints {
+			if isEndpointConnected(diskMap, s.endpointStrings[epIdx]) {
 				continue
 			}
-			disk, format, err := connectEndpoint(endpoint)
-			if err != nil {
-				printEndpointError(endpoint, err)
-				continue
+
+			connectQ <- struct{}{}
+
+			diskMapMu.RLock()
+			if len(diskMap) > len(s.endpoints)/2 {
+				diskMapMu.RUnlock()
+				return
 			}
-			i, j, err := findDiskIndex(s.format, format)
-			if err != nil {
-				// Close the internal connection to avoid connection leaks.
-				disk.Close()
-				printEndpointError(endpoint, err)
-				continue
-			}
-			disk.SetDiskID(format.XL.This)
-			s.xlDisks[i][j] = disk
-			onlineDisks++
+			diskMapMu.RUnlock()
+
+			go func(epIdx int) {
+				defer func() { <-connectQ }()
+				endpoint := s.endpoints[epIdx]
+				disk, format, err := connectEndpoint(endpoint)
+				if err != nil {
+					printEndpointError(endpoint, err)
+					return
+				}
+				i, j, err := findDiskIndex(s.format, format)
+				if err != nil {
+					// Close the internal connection to avoid connection leaks.
+					disk.Close()
+					printEndpointError(endpoint, err)
+					return
+				}
+				disk.SetDiskID(format.XL.This)
+
+				s.xlDisksMu.Lock()
+				s.xlDisks[i][j] = disk
+				s.xlDisksMu.Unlock()
+
+				diskMapMu.Lock()
+				diskMap[s.endpointStrings[epIdx]] = disk
+				diskMapMu.Unlock()
+			}(epIdx)
 		}
 		// Sleep for a while - so that we don't go into
 		// 100% CPU when half the disks are online.

--- a/cmd/xl-sets.go
+++ b/cmd/xl-sets.go
@@ -116,6 +116,25 @@ func isEndpointConnected(diskMap map[string]StorageAPI, endpoint string) bool {
 	return disk.IsOnline()
 }
 
+func (s *xlSets) getOnlineDisksCount() int {
+	s.xlDisksMu.RLock()
+	defer s.xlDisksMu.RUnlock()
+	count := 0
+	for i := 0; i < s.setCount; i++ {
+		for j := 0; j < s.drivesPerSet; j++ {
+			disk := s.xlDisks[i][j]
+			if disk == nil {
+				continue
+			}
+			if !disk.IsOnline() {
+				continue
+			}
+			count++
+		}
+	}
+	return count
+}
+
 func (s *xlSets) getDiskMap() map[string]StorageAPI {
 	diskMap := make(map[string]StorageAPI)
 
@@ -182,54 +201,11 @@ func findDiskIndex(refFormat, format *formatXLV3) (int, int, error) {
 // connectDisksWithQuorum is same as connectDisks but waits
 // for quorum number of formatted disks to be online in any given sets.
 func (s *xlSets) connectDisksWithQuorum() {
-	diskMap := make(map[string]StorageAPI)
-	var diskMapMu sync.RWMutex
-
-	connectQ := make(chan struct{}, len(s.endpoints)/2)
-
 	for {
-		for epIdx := range s.endpoints {
-			if isEndpointConnected(diskMap, s.endpointStrings[epIdx]) {
-				continue
-			}
-
-			connectQ <- struct{}{}
-
-			diskMapMu.RLock()
-			if len(diskMap) > len(s.endpoints)/2 {
-				diskMapMu.RUnlock()
-				return
-			}
-			diskMapMu.RUnlock()
-
-			go func(epIdx int) {
-				defer func() { <-connectQ }()
-				endpoint := s.endpoints[epIdx]
-				disk, format, err := connectEndpoint(endpoint)
-				if err != nil {
-					printEndpointError(endpoint, err)
-					return
-				}
-				i, j, err := findDiskIndex(s.format, format)
-				if err != nil {
-					// Close the internal connection to avoid connection leaks.
-					disk.Close()
-					printEndpointError(endpoint, err)
-					return
-				}
-				disk.SetDiskID(format.XL.This)
-
-				s.xlDisksMu.Lock()
-				s.xlDisks[i][j] = disk
-				s.xlDisksMu.Unlock()
-
-				diskMapMu.Lock()
-				diskMap[s.endpointStrings[epIdx]] = disk
-				diskMapMu.Unlock()
-			}(epIdx)
+		s.connectDisks()
+		if s.getOnlineDisksCount() > len(s.endpoints)/2 {
+			return
 		}
-		// Sleep for a while - so that we don't go into
-		// 100% CPU when half the disks are online.
 		time.Sleep(100 * time.Millisecond)
 	}
 }
@@ -314,13 +290,14 @@ const defaultMonitorConnectEndpointInterval = time.Second * 10 // Set to 10 secs
 // Initialize new set of erasure coded sets.
 func newXLSets(endpoints Endpoints, format *formatXLV3, setCount int, drivesPerSet int) (*xlSets, error) {
 	endpointStrings := make([]string, len(endpoints))
-	for _, endpoint := range endpoints {
+	for i, endpoint := range endpoints {
 		if endpoint.IsLocal {
-			endpointStrings = append(endpointStrings, endpoint.Path)
+			endpointStrings[i] = endpoint.Path
 		} else {
-			endpointStrings = append(endpointStrings, endpoint.String())
+			endpointStrings[i] = endpoint.String()
 		}
 	}
+
 	// Initialize the XL sets instance.
 	s := &xlSets{
 		sets:               make([]*xlObjects, setCount),

